modules/renter/repair.go

```diff
@@ -193,7 +194,11 @@ func (r *Renter) managedRepairIteration(rs *repairState) {
 		}
 
 		// Ignore workers that have had an upload failure recently.
-		if time.Since(worker.recentUploadFailure) < uploadFailureCooldown {
+		penalty := uint64(worker.consecutiveUploadFailures)
+		if worker.consecutiveUploadFailures > maxConsecutivePenalty {
+			penalty = uint64(maxConsecutivePenalty)
+		}
+		if time.Since(worker.recentUploadFailure) < uploadFailureCooldown*(1<<penalty) {
```

Comment:
no, because we let the consecutive failures count continue to increase even after we've reached the timeout. If we want to prevent overflows, we have to stop incrementing the consecutive failures after we've reached the maximum timeout.

The worker's job should not be to manage what time timeout is, that should be the uploaders job. The worker just reports the status and then the uploader decides which workers are useful. We don't want to cross the logic.